nbformat/notebooknode.py

```diff
@@ -2,9 +2,36 @@
 
 from ipython_genutils.ipstruct import Struct
 
+
 class NotebookNode(Struct):
     """A dict-like node with attribute-access"""
-    pass
+
+    def __setitem__(self, key, value):
+        """Set an item with check for allownew.
+        Examples
+        --------
+        >>> n = NotebookNode()
+        >>> n['a'] = 10
+        >>> n.allow_new_attr(False)
+        >>> n['a'] = 10
+        >>> n['a']
+        10
+        >>> try:
+        ...     n['b'] = 20
+        ... except KeyError:
+        ...     print('this is not allowed')
+        ...
+        this is not allowed
+        """
+        if not self._allownew and key not in self:
```

Comment:
Instead of re-implementing this check, we can use `super`:

```python
if isinstance(value, dict):
    value = from_dict(value)
return super(NotebookNode, self).__setitem__(key, value)
```